lib/matplotlib/tests/test_legend.py

```diff
@@ -1450,3 +1450,32 @@ def test_boxplot_labels():
     handles, labels = ax.get_legend_handles_labels()
     assert len(handles) == 0
     assert len(labels) == 0
+
+
+def test_boxplot_legend_labels():
+    # Test that legend entries are generated from boxplots. `label` is the new
+    # parameter for setting legend labels.
+    np.random.seed(19680801)
+    data = np.random.random((10, 4))
+    fig, axs = plt.subplots(nrows=1, ncols=3)
+
+    # Testing legend labels and patch passed to legend.
+    bp1 = axs[0].boxplot(data, patch_artist=True, label=['A', 'B', 'C', 'D'])
+    for i, v in enumerate(bp1['boxes']):
+        assert v.get_label() == ['A', 'B', 'C', 'D'][i]
+    handles, labels = axs[0].get_legend_handles_labels()
+    assert labels == ['A', 'B', 'C', 'D']
```

Comment:
Sorry, my previous recommendation to inline legend labels was not quite good as it's still used multiple times. It's reasonable to keep the variable, but only for the context of the single test.

```suggestion
    legend_labels = ['A', 'B', 'C', 'D']
    bp1 = axs[0].boxplot(data, patch_artist=True, label=legend_labels)
    for box, label in zip(bp1['boxes'], legend_labels):
        assert box.get_label() == label
    handles, labels = axs[0].get_legend_handles_labels()
    assert labels == legend_labels
```